packages/sdk/src/evm/common/any-evm-utils.ts

```diff
@@ -494,17 +498,83 @@ export async function getDeploymentInfo(
   create2Factory?: string,
 ): Promise<DeploymentInfo> {
   const compilerMetadata = await fetchPreDeployMetadata(metadataUri, storage);
+  let finalInfraContracts: InfraContractType[] = [];
+  let pluginTransactions: PrecomputedDeploymentTransaction[] = [];
+  let customParams: any[] = [];
+
   if (!create2Factory) {
     create2Factory = await getCreate2FactoryAddress(provider);
```

Comment:
create another const here, so the type is guaranteed to be `string`, or use an invariant - so you don't have to do `create2Factory as string` below which is error prone

Reply:
`const create2factoryAddress = create2Factory ? create2Factory : await getCreate2FactoryAddress(provider);`